src/tools/illink/test/Mono.Linker.Tests.Cases/DataFlow/AttributePropertyDataflow.cs

```diff
@@ -156,6 +161,40 @@ class ClassWithKeptPublicMethods
 			}
 		}
 
+		class AttributeWithConditionalExpression
+		{
+			[Kept]
+			[KeptAttributeAttribute (typeof (KeepsPublicMethodsAttribute))]
+			// Trimmer/NativeAot only for now - https://github.com/dotnet/linker/issues/2273
+			[ExpectedWarning ("IL2026", "--ClassWithKeptPublicMethods--", ProducedBy = Tool.Trimmer | Tool.NativeAot)]
+			[KeepsPublicMethods (TypeName = 1 + 1 == 2 ? "Mono.Linker.Tests.Cases.DataFlow.AttributePropertyDataflow+AttributeWithConditionalExpression+ClassWithKeptPublicMethods" : null)]
+			public static void Test ()
+			{
+				typeof (AttributeWithConditionalExpression).GetMethod ("Test").GetCustomAttribute (typeof (KeepsPublicMethodsAttribute));
+				typeof (AttributeWithConditionalExpression).GetField ("field").GetCustomAttribute (typeof (KeepsPublicMethodsAttribute));
+			}
+
+			// This testcase is an example where the analyzer may have a branch value while analyzing an attribute,
+			// where the owning symbol is not a method.
+			[Kept]
+			[KeptAttributeAttribute (typeof (KeepsPublicMethodsAttribute))]
+			// NativeAot doesn't handle the type name on fields: https://github.com/dotnet/runtime/issues/92259
+			[ExpectedWarning ("IL2105", "Mono.Linker.Tests.Cases.DataFlow.AttributePropertyDataflow+AttributeWithConditionalExpression+ClassWithKeptPublicMethods", ProducedBy = Tool.NativeAot)]
+			[ExpectedWarning ("IL2026", "--ClassWithKeptPublicMethods--", ProducedBy = Tool.Trimmer)]
+			[KeepsPublicMethods (TypeName = 1 + 1 == 2 ? "Mono.Linker.Tests.Cases.DataFlow.AttributePropertyDataflow+AttributeWithConditionalExpression+ClassWithKeptPublicMethods" : null)]
```

Comment:
Wow - didn't know it's possible to use expressions in attribute parameters... TIL :-)

Reply:
Only if they can be reduced to a constant during compilation time